limits/imap.go

```diff
@@ -10,63 +10,71 @@ import (
 
 // IMAP contains configurable upper limits that can be enforced by the Gluon server.
 type IMAP struct {
-	maxMailboxCount           int
-	maxMessageCountPerMailbox int
-	maxUIDValidity            int
-	maxUID                    int
+	maxMailboxCount           int64
+	maxMessageCountPerMailbox int64
+	maxUIDValidity            int64
+	maxUID                    int64
 }
 
 func (i IMAP) CheckMailBoxCount(mailboxCount int) error {
-	if mailboxCount >= i.maxMailboxCount {
+	if int64(mailboxCount) >= i.maxMailboxCount {
 		return ErrMaxMailboxCountReached
 	}
 
 	return nil
 }
 
 func (i IMAP) CheckMailBoxMessageCount(existingCount int, newCount int) error {
-	nextMessageCount := existingCount + newCount
+	nextMessageCount := int64(existingCount) + int64(newCount)
 
-	if nextMessageCount > i.maxMessageCountPerMailbox || nextMessageCount < existingCount {
+	if nextMessageCount > i.maxMessageCountPerMailbox || nextMessageCount < int64(existingCount) {
 		return ErrMaxMailboxMessageCountReached
 	}
 
 	return nil
 }
 
 func (i IMAP) CheckUIDCount(existingUID imap.UID, newCount int) error {
-	nextUIDCount := int(existingUID) + newCount
+	nextUIDCount := int64(existingUID) + int64(newCount)
 
-	if nextUIDCount > i.maxUID || nextUIDCount < int(existingUID) {
+	if nextUIDCount > i.maxUID || nextUIDCount < int64(existingUID) {
 		return ErrMaxUIDReached
 	}
 
 	return nil
 }
 
 func (i IMAP) CheckUIDValidity(uid imap.UID) error {
-	if int(uid) >= i.maxUIDValidity {
+	if int64(uid) >= i.maxUIDValidity {
 		return ErrMaxUIDValidityReached
 	}
 
 	return nil
 }
 
 func DefaultLimits() IMAP {
+	var maxInt int64
+	// Use MaxInt32 instead of MaxUint32 on 32 bit platforms
+	if uint64(^uintptr(0)) == ^uint64(0) {
```

Comment:
I'm not sure this will compile on 32bits. I just tried `if uint64(^uintptr(0)) == ^uint32(0) ` on my machine that failed to compile.

Reply:
It compiled fine on my 32 bit test VM. Anyway, changed to using `UintSize` from `math/bits`.